### DISCUSSION TITLE
A tool for fixing bad sectors manually...

### DISCUSSION THREAD

**Discussion by realmml** [keirf/disk-utilities#134] [Ideas]
Hi again! :)

This is a problem that I got while preserving Mot [(https://hol.abime.net/2520)](https://hol.abime.net/2520) a very rare spanish game edition.

In reality, every 16 bit game released in Spain is rare, because we had high pirate rates during the 16 bits era, this game maybe sold a hundred of units as maximum.

Then we preserved the game, but one sector had a crc error. Of course, we cleaned the floppy properly and made a lot of dumps until we reached the best dump possible of it. Because I know 68000 assembly and I got access to a crack of this game, I was able to discover that there was only one bit bad in the full sector.

Then my only option to fix the dump was really dirty (or I think that it is a really dirty way): write the dump in another floppy; fix the bit in that floppy using a sector editor; redump this new floppy and replace the bad sector in the original raw dump for the new one; and generating the new ipf/scp from the raw dump with the sector fixed.

I asked for any help to fix that bit in the cleanest way. And again the only option that it was given to me, it was to send the dump and they will fix the bit for me.

I don't know if it would be possible to make a tool that let you fix cases like that, where I can fix manually the bad bytes directly in the low level format. What do you think about it? Do you have any ideas?

Thanks for your time and patience!
Mauri.

**Comment by keirf** [comment]
Not really. Perhaps something could be cobbled together. What format are we talking about? IPF?

**Reply by d235j** [reply]
If you have a Mac handy you could perhaps bring the SCP or KFRaw files into the Applesauce software and fix the broken bit using its flux editor, which allows inserting, deleting, or pushing around specific fluxes. However I don't think there is a suitable format you can export to (yet).

What are the preferred formats for copy-protected Amiga at this point?

**Reply by realmml** [reply]
> Not really. Perhaps something could be cobbled together. What format are we talking about? IPF?

Well, the idea should be to fix the bits in the raw format, because after fixing the bad fluxes we could generate with disk-analyze the format that we want SCP, IPF, EADF, ...


> If you have a Mac handy you could perhaps bring the SCP or KFRaw files into the Applesauce software and fix the broken bit using its flux editor, which allows inserting, deleting, or pushing around specific fluxes. However I don't think there is a suitable format you can export to (yet).

Not Mac here, although I will ask to a friend to take a look to Applesauce.

> What are the preferred formats for copy-protected Amiga at this point?

I imagine that emulator users should prefer IPF or EADF (if it can handle the protection), but if you are trying to write the software in real floppies then some raw format.

**Reply by keirf** [reply]
So, editing a flux format like SCP or Kryoflux, ideally?

**Reply by realmml** [reply]
I think that editing fluxes for fixing bit errors should be the most useful.

But maybe is too difficult and it would be more practical a tool that let you convert the problematic track from fluxes to mfm or another human readable format, that it can be modified easily with an hexadecimal editor.

Then after fixing that mfm track, a tool will receive the original track in fluxes format and the mfm one for generating a new track of fluxes with the corrections applied.

And replacing the old track of fluxes for the new one, I can use disk-analyze for generating IPFs, ...

**Reply by d235j** [reply]
The Applesauce software can open, edit, and save flux data in A2R files. If other tools would add read/write support for A2R, this could be used to make the desired edits.

See #94.
